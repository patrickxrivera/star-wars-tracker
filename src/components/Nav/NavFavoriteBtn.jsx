import React from 'react';
import styled from 'styled-components';
import { Heart } from 'react-feather';

const Btn = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 100;
  width: 145px;
  height: 44px;
  background: #3e70ff;
  border: none;
  box-shadow: rgba(23, 43, 99, 0.26) 0 7px 42px;
  color: #fff;
  border-radius: 5px;
  letter-spacing: 0.5px;
  font-size: 12px;
  font-weight: bold;

  &:hover {
    opacity: 0.9;
    cursor: pointer;
  }

  &:focus {
    outline: 0;
  }
`;

const SmallHeart = styled(Heart)`
  width: 13px;
  height: 13px;
  padding: 0 4px;
`;

function NavFavoriteBtn() {
  return (
    <Btn>
      See Favorites <SmallHeart /> (0)
    </Btn>
  );
}

export default NavFavoriteBtn;
