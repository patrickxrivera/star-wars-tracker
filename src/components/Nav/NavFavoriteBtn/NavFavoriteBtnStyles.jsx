import { Link } from 'react-router-dom';
import styled from 'styled-components';

import { Heart } from 'react-feather';
import { getBgColorFrom } from '../../Results/ResultCard/helpers/resultCardStylesHelpers';

export const Btn = styled.button`
  z-index: 1;
  width: 145px;
  height: 44px;
  background: ${props => getBgColorFrom({ ...props }) || '#3e70ff'};
  border: none;
  color: #fff;
  border-radius: 5px;
  letter-spacing: 0.5px;
  font-size: 12px;
  font-weight: bold;

  &:hover {
    opacity: 0.9;
    cursor: pointer;
  }

  &:focus {
    outline: 0;
  }
`;

export const StyledLink = styled(Link)`
  text-decoration: none;
  color: #fff;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;

  &:focus,
  &:hover,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
  }
`;

export const SmallHeart = styled(Heart)`
  width: 13px;
  height: 13px;
  padding: 0 4.5px;
  stroke-width: 3;
`;
