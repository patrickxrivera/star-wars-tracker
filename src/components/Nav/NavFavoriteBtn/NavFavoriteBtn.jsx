import React from 'react';

import { Btn, SmallHeart } from './NavFavoriteBtnStyles.jsx';
import {
  getNumOf,
  getLocalStorageFor
} from '../../Results/ResultsContainer/helpers/resultsContainerHelpers';
import { StyledLink } from './NavFavoriteBtnStyles.jsx';

const NavFavoriteBtn = ({ selected }) => {
  const numOfFavorites = getNumOf('favorites');
  const cachedResults = getLocalStorageFor(selected);

  return (
    <Btn type={selected}>
      <StyledLink
        to={{
          pathname: '/results',
          state: { cachedResults, selected: 'favorites' }
        }}
      >
        See Favorites <SmallHeart /> ({numOfFavorites})
      </StyledLink>
    </Btn>
  );
};

export default NavFavoriteBtn;
