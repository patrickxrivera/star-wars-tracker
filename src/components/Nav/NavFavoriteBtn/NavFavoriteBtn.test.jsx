import React from 'react';
import { mount } from 'enzyme';
import NavFavoriteBtn from './NavFavoriteBtn.jsx';

test('NavFavoriteBtn renders correctly', () => {
  const wrapper = mount(<NavFavoriteBtn />);
  expect(wrapper).toMatchSnapshot();
});
