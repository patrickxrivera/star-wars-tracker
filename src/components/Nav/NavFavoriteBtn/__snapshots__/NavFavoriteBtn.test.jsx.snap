// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavFavoriteBtn renders correctly 1`] = `
<NavFavoriteBtn>
  <styled.button>
    <button
      className="sc-bdVaJa KOkiO"
    >
      See Favorites 
      <Styled(Heart)>
        <Heart
          className="sc-bwzfXH cxOpbI"
          color="currentColor"
          size="24"
        >
          <svg
            className="sc-bwzfXH cxOpbI"
            fill="none"
            height="24"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
            />
          </svg>
        </Heart>
      </Styled(Heart)>
       (0)
    </button>
  </styled.button>
</NavFavoriteBtn>
`;
